packages/reports/addon/templates/components/report-builder.hbs

```diff
@@ -17,35 +17,52 @@
         onAddDimension=(update-report-action "ADD_DIMENSION")
         onRemoveTimeGrain=(update-report-action "REMOVE_TIME_GRAIN")
         onRemoveDimension=(update-report-action "REMOVE_DIMENSION")
-        onToggleDimFilter=(update-report-action "TOGGLE_DIM_FILTER")
+        onToggleDimFilter=(update-report-action "TOGGLE_DIM_FILTER" (action (optional this.onUpdateCollapsed) false))
       }}
       {{metric-selector
         class=(concat "report-builder__container report-builder__metric-selector" (if disabled " report-builder__container--disabled" ""))
         request=request
         onAddMetric=(update-report-action "ADD_METRIC")
         onAddMetricWithParameter=(update-report-action "ADD_METRIC_WITH_PARAM")
         onRemoveMetric=(update-report-action "REMOVE_METRIC")
-        onToggleMetricFilter=(update-report-action "TOGGLE_METRIC_FILTER")
+        onToggleMetricFilter=(update-report-action "TOGGLE_METRIC_FILTER" (action (optional this.onUpdateCollapsed) false))
+        expandFilters=(action (optional this.onUpdateCollapsed) false)
       }}
     {{/if}}
   </div>
 </div>
 <div class="report-builder__main">
   {{#if hasValidLogicalTable}}
-    <div class={{concat "report-builder__container report-builder__container--filters" (if disabled " report-builder__container--disabled" "")}}>
+    <div class={{
+      concat "report-builder__container report-builder__container--filters"
+        (if this.disabled " report-builder__container--disabled")
+        (if this.isFiltersCollapsed " report-builder__container--filters--collapsed")
+      }}>
       <span class="report-builder__container-header">
-        <h3 class="report-builder__container-header__title">Filters</h3>
-        <span class="report-builder__container-header__message">
-          Click the icon {{navi-icon "filter" class="report-builder__container-header__message--icon"}} next to a dimension/metric to add a filter.
+        <span
+          class="report-builder__container-header__filters-toggle"
+          role="button"
+          onclick={{action (optional this.onUpdateCollapsed) (not this.isFiltersCollapsed)}}>
```

Comment:
I think this also works and we should try to use it going forward
```suggestion
          {{on "click" (fn (optional this.onUpdateCollapsed) (not this.isFiltersCollapsed))}}>
```

see [`on` rfc](https://github.com/emberjs/rfcs/blob/master/text/0471-on-modifier.md) and [`fn` rfc](https://github.com/emberjs/rfcs/blob/master/text/0470-fn-helper.md)